cmd/template/nodepool/flag.go

```diff
@@ -121,14 +130,26 @@ func (f *flag) Validate() error {
 	if f.NodepoolName == "" {
 		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagNodepoolName)
 	}
-	if f.NodesMax < 1 {
-		return microerror.Maskf(invalidFlagError, "--%s must be > 0", flagNodesMax)
-	}
-	if f.NodesMin < 1 {
-		return microerror.Maskf(invalidFlagError, "--%s must be > 0", flagNodesMin)
-	}
-	if f.NodesMin > f.NodesMax {
-		return microerror.Maskf(invalidFlagError, "--%s must be <= --%s", flagNodesMin, flagNodesMax)
+
+	{
+		// Validate scaling.
+		if f.NodesMax < 1 {
```

Comment:
Parameter description mentions default values for these, but how does defaulting work compared to this validation?

Reply:
I moved the defaulting here, so I could have different defaults for each provider https://github.com/giantswarm/kubectl-gs/blob/6df4611abf93eeb5be9d54f43c87c82c1df7a5cc/cmd/template/nodepool/runner.go#L39

So basically, if the `--provider` flag is `aws`, then the defaults are `min 3` and `max 10`, but if the flag is `azure`, then the defaults are `min 3` and `max 3`

Reply:
Ah, right. `Validate()` gets called _after_ that defaulting. I missed that earlier. :+1: